tests/vlan/test_autostate_disabled.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import pytest
+
+from tests.common.utilities import wait_until
+
+
+pytestmark = [
+    pytest.mark.topology("t0")
+]
+
+
+class TestAutostateDisabled:
+    """
+    This test case is used to verify that vlan interface autostate is **disabled** on SONiC.
+
+    The autostate feature notifies a switch or routing module VLAN interface (Layer 3 interface) to transition to
+    up/up status when at least one Layer 2 port becomes active in that VLAN.
+
+    In SONiC, all vlans are bound to a single bridge interface, so the vlan interface will go down only if the bridge
+    is down. Since bridge goes down when all the associated interfaces are down, if all the vlan members across all
+    the vlans go down, the bridge will go down and the vlan interface will go down.
+
+    For more information about autostate, see:
+      * https://www.cisco.com/c/en/us/support/docs/switches/catalyst-6500-series-switches/41141-188.html
+    """
+
+    def test_autostate_disabled(self, duthosts, enum_frontend_dut_hostname):
+        """
+        Verify vlan interface autostate is disabled on SONiC.
+        """
+
+        duthost = duthosts[enum_frontend_dut_hostname]
+        dut_hostname = duthost.hostname
+
+        # Collect DUT configuration and status
+        vlan_members_facts = duthost.get_running_config_facts().get('VLAN_MEMBER')
+        if vlan_members_facts is None:
+            pytest.skip('No vlan available on DUT {hostname}'.format(hostname=dut_hostname))
+        ifs_status = self.get_interface_status(duthost)
+        ip_ifs = duthost.show_ip_interface()['ansible_facts']['ip_interfaces']
+
+        # Find out all vlans which meet the following requirements:
+        #   1. The oper_state of vlan interface is 'up'
+        #   2. The oper_state of at least one member in the vlan is 'up'
+        vlan_available = []
+        for vlan in vlan_members_facts:
+            if ip_ifs.get(vlan, {}).get('oper_state') == 'up':
+                for member in vlan_members_facts[vlan]:
+                    if ifs_status.get(member, {}).get('oper') == 'up':
+                        vlan_available.append(vlan)
+                        break
+        if len(vlan_available) == 0:
+            pytest.skip('No vlan available on DUT {hostname}'.format(hostname=dut_hostname))
+
+        # Pick a vlan for test
+        vlan = vlan_available[0]
+        vlan_members = vlan_members_facts[vlan].keys()
+
+        # Shutdown all the members in vlan.
+        res = duthost.shutdown_multiple(vlan_members)
+        if not res.is_successful:
+            self.restore_interface_admin_state(duthost, ifs_status)
+            pytest.fail('shutdown "{vlan_members}" in {vlan} failed'.format(vlan_members=vlan_members, vlan=vlan))
```

Comment:
You can use this pattern, then you don't have to write the recover code everywhere.
```python
try:
   # test code
finally:
   self.restore...()
```

And please make sure intf's oper states are recovered before leave this test

Reply:
Thanks, I have fixed this in latest commit.